internal/file/file.go

```diff
@@ -110,3 +116,34 @@ func CopyDirToDir(src, dst string) error {
 func TrimFileExtension(fileName string) string {
 	return strings.TrimSuffix(fileName, filepath.Ext(fileName))
 }
+
+// WriteFile writes content to path with all parent directories created.
+// If path already exists and is a file, WriteFile overwrites it.
+func WriteFile(path string, content []byte) (writeErr error) {
+	if err := os.MkdirAll(filepath.Dir(path), 0700); err != nil {
+		return err
+	}
```

Comment:
It seems not necessary to call `MkdirAll` each time `WriteFile`, especially when you have this logic at `crl.go:71`.

Reply:
This method is under internal/file at the moment, which is shared by packages. The [os.WriteFile](https://pkg.go.dev/os#WriteFile) has the similar behavior though. 
If we remove `MkdirAll`, caller has to create the dir before writing every time.
An alternative would be moving this function back to `crl` package. What do you think? @shizhMSFT

Reply:
Updated.